mlx/primitives.h

```diff
@@ -2158,4 +2158,23 @@ class Cholesky : public UnaryPrimitive {
   bool upper_;
 };
 
+class Eigvalsh : public UnaryPrimitive {
+ public:
+  explicit Eigvalsh(Stream stream, bool upper, bool compute_vectors)
```

Comment:
Given it can compute eigenvectors, this can probably be called `Eigh` rather than `Eigvalsh`?